sycl/doc/extensions/experimental/sycl_ext_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -0,0 +1,929 @@
+a small 2-dimensional matrix that supports native operations in
+hardware. There are a number of template parameters, namely the group
+scope, the type of the elements, the matrix use, the shape, and the
+memory layout of the matrix.  This results in the following description:
+
+```c++
+namespace sycl::ext::oneapi::experimental::matrix {
+
+template <typename Group, typename T, use Use, size_t Rows, size_t Cols,
+          layout Layout = (Use == use::accumulator) ?
+                          layout::dynamic : /*unspecified*/ >
+struct joint_matrix {
+  joint_matrix();
+  joint_matrix(const joint_matrix &) = delete;
+  joint_matrix &operator=(const joint_matrix &) = delete;
+};
+
+} // namespace sycl::ext::oneapi::experimental::matrix
+```
+The constructor for the `joint_matrix` type is a group function as
+defined in section 4.17.3 of the core SYCL specification. It must be
+encountered in converged control flow by all work-items in the
+`Group`.
+
+==== Group Memory Scope
+Most operations on the joint_matrix are group functions, meaning that
+all work items in a group collectively perform an operation on the
+same matrix. The `Group` template parameter specifies the execution
+scope of the work-items in the group. The `joint_matrix` is shared among the
+work items in the group and is not private to each work item. This
+extension currently supports only the sub-group scope, so the `Group`
+template parameter must be `sycl::sub_group`, and group operations for
+the joint matrix must be done collectively by the work-items in a
+single sub-group. In this case, a matrix is declared as follows:
+
+```c++
+joint_matrix<sub_group, int8_t, use::a, tM, tN, layout::row_major> tA;
+```
+
+==== Element Type
+The `T` template parameter specifies the type of each element in the
+matrix. Each device supports only certain element types, so the
+application must use the query operations (defined below) to ensure
+that the element type is supported on the device where the kernel
+using this `joint_matrix` runs.
+
+==== Matrix Use
+The main operation performed by the matrix hardware is `D=C+A*B`. The
+`Use` template parameter specifies which of these terms (A, ,B, C, or D)
+corresponds to the `joint_matrix` object. The use enumeration defines
+the set of legal values. The A matrix must have the value `use::a`. The
+B matrix must have the value `use::b`. The C and D matrices must both
+have the value `use::accumulator`. This is used by backend
+implementations to reason about the layout of the matrix in
+registers.
+
+```c++
+namespace sycl::ext::oneapi::experimental::matrix {
+
+enum class use {
+  a,
+  b,
+  accumulator
+};
+
+} // namespace sycl::ext::oneapi::experimental::matrix
+```
+
+==== Matrix Shape
+The `Rows` and `Cols` template parameters tell the number of rows and
+columns in the joint matrix. Each device supports only certain
+combinations of row and column sizes, so the application must use the
```

Comment:
```suggestion
combinations of row and column sizes, so the application can use the
```

Reply:
Same here.  Isn't "must" better?

Reply:
I think you could resolve this issue using similar wording to what you added for "Element type":

> The `Rows` and `Cols` template parameters provide the number of rows and columns in the joint matrix. Each device supports only certain combinations of row and column sizes, so the application must ensure that the combination is supported on the device where the kernel using this `joint_matrix` runs. The query functions (defined below) may be used to determine the set of combinations that are supported on a device.